components/push/android/src/main/java/org/mozilla/push/PushConnection.kt

```diff
@@ -130,168 +112,73 @@ class PushConnection(path: String, encryption_key: String? = null) : PushAPI, Au
-     * /
-    fun noteObservation(data: VisitObservation)
-
-    /**
-     * A way to search the internal database tailored for autocompletion purposes.
+     * Get the Subscription Info block
      *
-     * @param query a string to match results against.
-     * @param limit a maximum number of results to retrieve.
-     * @return a list of [SearchResult] matching the [query], in arbitrary order.
-     * /
-    fun queryAutocomplete(query: String, limit: Int): List<SearchResult>
+     * @param channelID Channel ID (UUID) for new subscription
+     * @return a Subscription Info structure
+     */
+    fun getSubscriptionInfo(
+        channelID: String
+    ): SubscriptionInfo
 
     /**
-     * Maps a list of page URLs to a list of booleans indicating if each URL was visited.
-     * @param urls a list of page URLs about which "visited" information is being requested.
-     * @return a list of booleans indicating visited status of each
-     * corresponding page URI from [urls].
-     * /
-    fun getVisited(urls: List<String>): List<Boolean>
+     * Unsubscribe a given channelID
+     *
+     * @param channelID Channel ID (UUID) for subscription to remove.
+     * @return bool.
+     */
+    fun unsubscribe(channelID: String): Boolean
 
     /**
-     * Returns a list of visited URLs for a given time range.
-     *
-     * @param start beginning of the range, unix timestamp in milliseconds.
-     * @param end end of the range, unix timestamp in milliseconds.
-     * @param includeRemote boolean flag indicating whether or not to include remote visits. A visit
-     *  is (roughly) considered remote if it didn't originate on the current device.
-     * /
-    fun getVisitedUrlsInRange(start: Long, end: Long = Long.MAX_VALUE, includeRemote: Boolean = true): List<String>
+     * Updates the Native OS push registration ID.
+     * @param registrationToken the new Native OS push registration ID.
+     * @return bool
+     */
+    fun update(registrationToken: String): Boolean
 
     /**
-     * Syncs the history store.
-     *
-     * Note that this function blocks until the sync is complete, which may
-     * take some time due to the network etc. Because only 1 thread can be
-     * using a PushAPI at a time, it is recommended, but not enforced, that
-     * you use a separate PushAPI instance purely for syncing.
+     * Verifies the connection state. NOTE: If the internal check fails,
+     * endpoints will be re-registered and new endpoints will be returned for
+     * known ChannelIDs
      *
-     * /
-    fun sync(syncInfo: SyncAuthInfo)
-}
-
-open class PushException(msg: String): Exception(msg)
-open class InternalPanic(msg: String): PushException(msg)
-open class UrlParseFailed(msg: String): PushException(msg)
-open class InvalidPlaceInfo(msg: String): PushException(msg)
-open class PushConnectionBusy(msg: String): PushException(msg)
+     * @return Map of ChannelID: Endpoint, be sure to notify apps registered to given
```

Comment:
Would it be expected that the android component calls this check before performing any action first?

Reply:
Not necessarily. This check can be a bit expensive, since it fetches content from the server and iterates through known connections. It's more a general "sanity check" on the known data state. Calling it once a day should be sufficient. 

In the past we've had a few occasions where a client has gotten out of sync with the server. This has resulted in the subscription provider never knowing that a given subscription is no longer valid and sending effectively useless data to the Push Service. This function provides a way to rectify that situation in the future.